### ISSUE TITLE
Segment undefined in the context of a middleware

### ISSUE THREAD

**Issue by sandoche** [dansmaculotte/nuxt-segment#13]
Hi,
I am trying to use segment from the context, in a middleware loaded from a layout and it does not work.

Segment undefined in the context in a middleware

```js
export default function({ app, $segment }) {
  console.log($segment)
  console.log(app.$segment)
}
```

**Comment by gaelreyrol** [comment]
Hi @sandoche !

Can you provide your configuration ?

**Comment by sandoche** [comment]
In nuxt.config.js

```js

modules: [
    // ...
    '@dansmaculotte/nuxt-segment'
  ],

  segment: {
    writeKey: process.env.SEGMENT_WRITE_KEY,
    disabled: false,
    useRouter: true
  }

```

When I use it in a page in mounted with this.$segment it does work.

**Comment by sandoche** [comment]
I am using Nuxt.js @ v2.14.7

**Comment by gaelreyrol** [comment]
Do you any error logs ?

**Comment by sandoche** [comment]
When I run a page with this middleware of a layout I have this:
`undefined` (two times)

No other errors.
When I use `this.$segment` in the mounted of a page it works perfectly.

**Comment by sandoche** [comment]
I tried to move the middleware in the page instead of the layout and it's the same

**Comment by gaelreyrol** [comment]
So it's working in a page but not in a middleware or a layout ?

**Comment by sandoche** [comment]
Exactly. It works in the mounted of the page. It also works automatically also for each page.

It does not work as a middleware (either for the page of the layout)

**Comment by gaelreyrol** [comment]
Okay, I will try to look at it soon !

**Comment by sandoche** [comment]
Thanks for now, I'll run the code on the mounted of the layout, it seems that this works. But I would have a cleaner code by having it in a middleware.
